paper_search_mcp/server.py

```
@@ -481,6 +496,91 @@ async def download_iacr(paper_id: str, save_path: str = "./downloads") -> str:
+        """Download a PDF via Sci-Hub using DOI, PMID, URL, or direct PDF URL.
+
+        Args:
+            identifier: DOI, PMID, article URL, or direct PDF URL.
+            save_path: Directory to save the PDF (default: './downloads').
+        Returns:
+            Path to the downloaded PDF file, or an error message on failure.
+        """
+        try:
+            fetcher = SciHubFetcher(output_dir=save_path)
+            result = fetcher.download_pdf(identifier)
+            return result if result else "Failed to download PDF from Sci-Hub."
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)